### PR TITLE
Suite tests

### PR DESCRIPTION
**Pull request by umbernhard**
**Description**

Adds tests to get SUITE up to 100% test coverage. Also removes unreachable code. 

**Testing**

Added testing for all the corner cases. 

**Progress**

Ready to go.

### CODE REVIEW

File: server/audit_math/suite.py [hunk 1]
```diff
@@ -102,9 +102,11 @@ def compute_pvalue(
         """
 
         n = self.sample_size
+        if n == 0 or reported_margin == 0:
+            return 1.0
 
-        if n == 0:
-            return 1
+        if self.sample_size == self.num_ballots:
+            return 0.0
```
Comment (jonahkagan) [inline server/audit_math/suite.py:109]: maybe just use `self.sample_size` everywhere and remove `n` for readability?

File: server/audit_math/suite.py [hunk 2]
```diff
@@ -133,17 +135,14 @@ def compute_pvalue(
...
+        upper_n_w_limit = (self.num_ballots - n_u + null_margin) / 2.0
         lower_n_w_limit = np.max([n_w, n_l + null_margin])
 
         # For extremely small or large null_margins, the limits do not
         # make sense with the sample values.
         if upper_n_w_limit < n_w or (upper_n_w_limit - null_margin) < n_l:
             raise ValueError("Null is impossible, given the sample")
 
-        if lower_n_w_limit > upper_n_w_limit:
-            lower_n_w_limit, upper_n_w_limit = upper_n_w_limit, lower_n_w_limit
```
Comment (jonahkagan) [inline server/audit_math/suite.py:145]: how are you sure this is unreachable?

File: server/audit_math/suite.py [hunk 3]
```diff
@@ -153,13 +152,6 @@ def compute_pvalue(
             )
         )
 
-        # Sometimes the upper_n_w_limit is too extreme, causing illegal 0s.
-        # Check and change the limit when that occurs.
-        if np.isinf(null_logLR(upper_n_w_limit)) or np.isinf(
-            LR_derivative(upper_n_w_limit)
-        ):
-            upper_n_w_limit -= 1
```
Comment (jonahkagan) [inline server/audit_math/suite.py:161]: same q: how do you know it's unreachable?

File: server/audit_math/suite.py [hunk 4]
```diff
@@ -172,10 +164,11 @@ def compute_pvalue(
             nuisance_param = sp.optimize.brentq(
                 LR_derivative, lower_n_w_limit, upper_n_w_limit
             )
+        # print(nuisance_param)
```
Comment (jonahkagan) [inline server/audit_math/suite.py:167]: remove

File: server/audit_math/suite.py [hunk 5]
```diff
@@ -534,24 +532,27 @@ def try_n(
         # Otherwise use the sample we've seen so far
         hyp_sample["hyp_round"] = {
             winner: min(
-                math.ceil(
-                    ((n2 - hyp_sample_size) * cumulative_sample[winner])
-                    / hyp_sample_size
+                int(
```
Comment (jonahkagan) [inline server/audit_math/suite.py:535]: why `int` vs `math.ceil`?

**Review by jonahkagan — APPROVED** [review]
i'd still like to go through and review all the math, which i haven't done yet, so this is prob fine to merge. if you want to answer my questions now, feel free, or we can talk about them when i review the math

File: server/audit_math/suite.py (hunk 1 above)
Comment (umbernhard) [reply server/audit_math/suite.py:109]: Ah, yeah I mean to do that. Will clean up.

File: server/audit_math/suite.py (hunk 2 above)
Comment (umbernhard) [reply server/audit_math/suite.py:145]: Yes. Taken with the statement above it, it is mathematically impossible for one of these ifs to be true when the other isn't. Therefore, if we pass the first if, this one can never happen. I worked out the math, I can take a picture of my notes if you would like!

File: server/audit_math/suite.py (hunk 3 above)
Comment (umbernhard) [reply server/audit_math/suite.py:161]: upper_n_w_limit is guaranteed to be a finite number. The only way for it to become infinite is if we do something like divide it by really, really small numbers. If you look in these two functions, that can't ever happen. Again, I worked this out on paper (and with a coding example.

File: server/audit_math/suite.py (hunk 5 above)
Comment (umbernhard) [reply server/audit_math/suite.py:535]: we want our sample to always be conservative, so we want to round votes for the winner down, not up. This was actually an error I had made in my original implementation. I'm using int because we have to do the cast anyways, and `int(math.floor(`is redundant.

Comment (jonahkagan) [reply server/audit_math/suite.py:533]: looks like `int` and `math.floor` are slightly different, fyi. dunno if it matters here though: https://stackoverflow.com/questions/31036098/what-is-the-difference-between-int-and-floor-in-python-3

Comment (umbernhard) [reply server/audit_math/suite.py:533]: hmm, tbh truncation probably saves us more headaches. So I think it's fine.
